src/python/era_5g_object_detection_distributed_interface/era_5g_object_detection_distributed_interface/interface.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+
+import logging
+import secrets
+from queue import Queue
+from era_5g_object_detection_distributed_interface.results_reader import ResultsReader
+
+import flask_socketio
+from era_5g_object_detection_distributed_interface.task_handler_gstreamer_rabbitmq_redis import \
+    TaskHandlerGstreamerRabbitmqRedis as TaskHandler
+from era_5g_netapp_interface.common import get_logger
+from flask import Flask, Response, request, session
+
+from flask_session import Session
+
+# flask initialization
+app = Flask(__name__)
+app.secret_key = secrets.token_hex()
+app.config['SESSION_TYPE'] = 'filesystem'
+
+# socketio initialization for sending results to the client
+Session(app)
+socketio = flask_socketio.SocketIO(app, manage_session=False, async_mode='threading') 
+
+# list of available ports for gstreamer communication - they needs to be exposed when running in docker / k8s
+free_ports = [5001, 5002, 5003]
+
+# list of registered tasks
+tasks = dict()
+
+# queue with all celery jobs for results retrieval
+jobs_info_queue = Queue(1024)
+
+logger = get_logger(log_level=logging.INFO)
+
+@app.route('/register', methods=['GET'])
+def register():
+    """
+    Needs to be called before an attempt to open WS is made.
+
+    Returns:
+        _type_: The port used for gstreamer communication.
+    """
+    global logger
```

Comment:
@Kapim `global` should not be needed here (only if you want to assign something).

Reply:
removed in multiple locations, thanks